File: algorithms/leet.0118.src.1.java
public class Solution {
    public List<List<Integer>> generate(int numRows) {
        
        // case == 0
        if(numRows == 0) return new ArrayList<List<Integer>>();
        
        // case > 0
        List<List<Integer>> ans = new ArrayList<List<Integer>>();
        ArrayList<Integer> last = new ArrayList<Integer>();
        last.add(1);
        
        for(int i = 0; i < numRows; ++i) {
            ans.add(last);
            ArrayList<Integer> current = new ArrayList<Integer>();
            current.add(1);
            for(int j = 1; j < last.size(); ++j) {
                current.add(last.get(j-1) + last.get(j));
            }
            current.add(1);
            last = current;
        }
        return ans;
    }
}

File: algorithms/leet.0118.src.1.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        ans = [[1]]
        for i in range(numRows-1):
            ans.append([0] + ans[-1])
            for j in range(i+1):
                ans[-1][j] += ans[-1][j+1]
        return ans
